### ISSUE TITLE
Terraform - Downloading GPG Key step gets stuck

### ISSUE THREAD

**Issue by DevSecNinja** [devcontainers/features#323]
Hello,

Since a few days I notice that the GPG Key download step in the Terraform feature hangs:

````
=> [internal] load metadata for mcr.microsoft.com/devcontainers/base:j  608.5s
 => => extracting sha256:9a586f3d84de83b25cb9ca6d0e733d37d5283da35a837917  0.8s
 => => extracting sha256:5ac951de24f647413f348bfa4183b92c9962b167e0a3106d  0.3s
 => => extracting sha256:1aa90a4f5c993fd6d1e5f0885a1794ee59253ee8d35463ea  0.0s
 => => extracting sha256:074fbd7f2200028d96665f87580813cabf3361b85a4217ce  0.0s
 => => extracting sha256:ee36e8b50d1a0a2f6169666f26f3938fe1ea3efabfb6df01  0.0s
 => => extracting sha256:9e8c9f9ef24f45765c7592d55a952cbe2d0d4fefcd797237  5.5s
 => [dev_containers_feature_content_normalize 2/3] COPY --from=dev_contai  4.8s
 => [dev_containers_feature_content_normalize 3/3] RUN chmod -R 0700 /tmp  1.7s
 => [dev_containers_target_stage  2/12] COPY --from=dev_containers_featur  0.3s
 => [dev_containers_target_stage  3/12] RUN echo "_CONTAINER_USER_HOME=$(  0.7s
 => [dev_containers_target_stage  4/12] RUN cd /tmp/build-features/azure  61.0s
 => [dev_containers_target_stage  5/12] RUN cd /tmp/build-features/github  6.1s
 => [dev_containers_target_stage  6/12] RUN cd /tmp/build-features/kubec  51.5s
 => [dev_containers_target_stage  7/12] RUN cd /tmp/build-features/power  34.3s
 => [dev_containers_target_stage  8/12] RUN cd /tmp/build-features/terr  431.1s
 => => # Downloading terraform...
 => => # TERRAFORM_GPG_KEY=72D7468F
 => => # GPG_KEY_SERVERS=keyserver hkp://keyserver.ubuntu.com:80
 => => # keyserver hkps://keys.openpgp.org
 => => # keyserver hkp://keyserver.pgp.com
 => => # (*) Downloading GPG key...
````

Screenshot:

![image](https://user-images.githubusercontent.com/14926452/205906767-81e2cef8-2c58-4ae8-b080-d9843ad918e8.png)

I have tried this on various systems/connections and it seems to consistently hang. As you can see, other features install fine.

Any idea how to fix this? Thanks!

**Comment by SoloW1ngPixy** [comment]
I have the same issue. 
Close remote container on VSCode and choose to rebuild container seem to work, but not very convenient

**Comment by sarcasmboy** [comment]
I am having the same issue.  Closing the remote container in VSCode and rebuilding the container did not solve my issue

**Comment by betancourtca** [comment]
Same issue with the Python feature getting stuck downloading PGP key

**Comment by betancourtca** [comment]
Same issue with the Python feature getting stuck on downloading PGP key

**Comment by sarcasmboy** [comment]
Here's some additional details.  In addition, I have run in a vanilla ubuntu devcontainer without the terraform feature installed and it hangs when executing `gpg --verbose --recv-keys 72D7468F` which is what appears to be what is executed in the [install.sh](https://github.com/devcontainers/features/blob/e28358e33b98998e00c2c62d5d561626d96bed24/src/terraform/install.sh#L80) script during setup.

I am able to reliably reproduce the issues, so if there are any additional debug steps you'd like me to take, let me knowl.

**System Information**
```
Hardware
Processor	12th Gen Intel(R) Core(TM) i7-12700H   2.30 GHz
Installed RAM	32.0 GB (31.7 GB usable)

OS
System type	64-bit operating system, x64-based processor
Edition		Windows 11 Enterprise
Version		22H2
Installed on	‎12/‎4/‎2022
OS build	22621.819
Experience	Windows Feature Experience Pack 1000.22636.1000.0

Docker
Docker Desktop	4.15.0 (93002) 
Backend		WSL 2
```

**devcontainer.json**
```json
// For format details, see https://aka.ms/devcontainer.json. For config options, see the
// README at: https://github.com/devcontainers/templates/tree/main/src/ubuntu
{
	"name": "Ubuntu",
	"image": "mcr.microsoft.com/devcontainers/base:jammy",
	"features": {
		"ghcr.io/devcontainers/features/terraform:1": {}
	}

	// Features to add to the dev container. More info: https://containers.dev/features.
	// "features": {},

	// Use 'forwardPorts' to make a list of ports inside the container available locally.
	// "forwardPorts": [],

	// Use 'postCreateCommand' to run commands after the container is created.
	// "postCreateCommand": "uname -a",

	// Configure tool-specific properties.
	// "customizations": {},

	// Uncomment to connect as root instead. More info: https://aka.ms/dev-containers-non-root.
	// "remoteUser": "root"
}
```

**Comment by samruddhikhandale** [comment]
I couldn't reproduce the issue with the dev config mentioned in https://github.com/devcontainers/features/issues/323#issuecomment-1340187092 😓  I've tried multiple times on a Windows and MacOS machine, every time the devcontainer is created/opened fine.

@sarcasmboy Could you provide me information on your Dev Container Extension version & VS Code version?

It is strange that in the original post, the container was able to download gpg keys for the `github-cli` Feature but couldn't for the `terraform` Feature 🤔 

@joshspicer / @edgonmsft any thoughts on how to fix this? or could you reproduce?

**Comment by sarcasmboy** [comment]
vscode details

```
Version: 1.73.1 (user setup)
Commit: 6261075646f055b99068d3688932416f2346dd3b
Date: 2022-11-09T04:27:29.066Z
Electron: 19.0.17
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.22621
Sandboxed: No
```

`code --list-extensions --show-versions`

```
ms-vscode-remote.remote-containers@0.266.1
ms-vscode-remote.remote-ssh@0.92.0
ms-vscode-remote.remote-ssh-edit@0.84.0
ms-vscode.remote-explorer@0.0.3
```

**Comment by samruddhikhandale** [comment]
I have the same exact configuration. Still can't reproduce.

@sarcasmboy by any change do you have gpg installed on your machine?

I wonder if this could be an issue?  🤔 
https://wiki.archlinux.org/title/GnuPG#gpg_hanged_for_all_keyservers_(when_trying_to_receive_keys)

![image](https://user-images.githubusercontent.com/24955913/206805512-eddeec97-4c10-4e93-9885-7b3cd5b9ac69.png)

**Comment by vnextcoder** [comment]
I have same issue. Below is `devcontainer.json` 

and it stucks only at Terraform Stage.

````
{
	"build": {
		"dockerfile": "./Dockerfile",
		"context": "."
	},
	"features": {
		"ghcr.io/devcontainers/features/common-utils:1": {
			"installZsh": "true",
			"username": "vscode",
			"uid": "1000",
			"gid": "1000",
			"upgradePackages": "true"
		},
		"ghcr.io/devcontainers/features/git:1": {
            "version": "latest",
            "ppa": "false"
        },
        "ghcr.io/devcontainers/features/azure-cli:1": {
            "version": "latest"
          },
          "ghcr.io/devcontainers/features/terraform:1": {
            "version":"1.1.0",
            "tflint":"0.43.0",
            "tfgrunt":"0.42.3"
          }

	},
	"remoteUser": "vscode"
}
````

**Comment by sarcasmboy** [comment]
@samruddhikhandale I copied the [install.sh](https://github.com/devcontainers/features/blob/main/src/terraform/install.sh) script into my repository and ran it within a vanilla `mcr.microsoft.com/devcontainers/base:jammy` devcontainer.  When I kill the dirmngr process during the hang, I see the script render an error and then retry until it gives up

<details>
  <summary>output</summary>

```
TERRAFORM_VERSION=1.3.6
TFLINT_VERSION=0.43.0
TERRAGRUNT_VERSION=0.42.3
Downloading terraform...
TERRAFORM_GPG_KEY=72D7468F
GPG_KEY_SERVERS=keyserver hkp://keyserver.ubuntu.com:80
keyserver hkps://keys.openpgp.org
keyserver hkp://keyserver.pgp.com
(*) Downloading GPG key...
gpg: keyserver receive failed: End of file
(*) Failed getting key, retring in 10s...
(*) Downloading GPG key...
gpg: keyserver receive failed: End of file
(*) Failed getting key, retring in 10s...
(*) Downloading GPG key...
gpg: keyserver receive failed: End of file
(*) Failed getting key, retring in 10s...
(*) Downloading GPG key...
gpg: keyserver receive failed: End of file
(*) Failed getting key, retring in 10s...
(*) Downloading GPG key...
gpg: keyserver receive failed: End of file
(*) Failed getting key, retring in 10s...
(!) Failed to get gpg key.
```

</details>

**Comment by edgonmsft** [comment]
I've been trying to think here but without a repro not sure what the issue could be, does it maybe repro localy with Devcontainers extension?

**Comment by vnextcoder** [comment]
There are so many people able to reproduce this issue. Even running install.sh on plain vanilla container in interactive mode renders same error

**Comment by chrmarti** [comment]
The feature script succeeds for me. Using the key id `gpg --verbose --recv-keys 72D7468F` fails. Using the fingerprint from https://www.hashicorp.com/security `gpg --verbose --recv-keys "C874 011F 0AB4 0511 0D02 1055 3436 5D94 72D7 468F"` on the other hand succeeds.

Maybe something with the lookup on the key servers changed. E.g., https://keys.openpgp.org/search?q=72D7468F complains that "Search by Short Key ID is not supported.". Not sure why the feature script still succeeds.

**Comment by austindimmer** [comment]
I am observing a similar issue trying to install Python. I tried originally via Python install script which has worked for me for 1+ years. The script was hanging at "Downloading GPG key..." so I migrated to use the feature instead and now I am observing that the Python feature is hanging at "Downloading GPG key..."

![image](https://user-images.githubusercontent.com/533463/207912871-2dadc339-dc67-481c-930c-3be124abf24a.png)

**Comment by samruddhikhandale** [comment]
@sarcasmboy Can you append `--verbose` flag to the install script in [here](https://github.com/devcontainers/features/blob/main/src/terraform/install.sh#L84)? Wondering if that might provide error details.

> ( echo "${keys}" | xargs -n 1 gpg --verbose -q ${keyring_args} --recv-keys) 2>&1 && gpg_ok="true"

**Comment by daniel-klepacki-medius** [comment]
I have the same issue for several days now.

**Comment by sarcasmboy** [comment]
@samruddhikhandale No change in the output

<details>
  <summary>output</summary>

```
TERRAFORM_VERSION=1.3.6
TFLINT_VERSION=0.43.0
TERRAGRUNT_VERSION=0.42.5
Downloading terraform...
TERRAFORM_GPG_KEY=72D7468F
GPG_KEY_SERVERS=keyserver hkp://keyserver.ubuntu.com:80
keyserver hkps://keys.openpgp.org
keyserver hkp://keyserver.pgp.com
(*) Downloading GPG key...
^C
gpg: signal 2 caught ... exiting
```

</details>

I also added some additional outputs to help debug

```bash
echo "(*) Downloading GPG key..."
echo "${keyring_args}"
echo "${keys}"
( echo "${keys}" | xargs -n 1 gpg --verbose -q ${keyring_args} --recv-keys) 2>&1 && gpg_ok="true"
```

<details>
  <summary>output</summary>

```
TERRAFORM_VERSION=1.3.6
TFLINT_VERSION=0.43.0
TERRAGRUNT_VERSION=0.42.5
Downloading terraform...
TERRAFORM_GPG_KEY=72D7468F
GPG_KEY_SERVERS=keyserver hkp://keyserver.ubuntu.com:80
keyserver hkps://keys.openpgp.org
keyserver hkp://keyserver.pgp.com
(*) Downloading GPG key...
keyring_args: 
keys: 72D7468F
^C  
gpg: signal 2 caught ... exiting
```

</details>

**Comment by DevSecNinja** [comment]
I'm not able to reproduce the issue anymore with my own devcontainer and the sample one mentioned by @sarcasmboy. Closing the issue for now, but feel free to comment/reopen if it doesn't work for you.

**Comment by eitamal** [comment]
I'm still getting the same issue, unfortunately.

But I've played around with it and discovered that the issue for me was **port 80** on the **ubuntu keyserver**.

`/tmp/tmp-gnupg/dirmngr.conf`:

```
disable-ipv6
log-file /tmp/tmp-gnupg/dirmngr.log
keyserver hkp://keyserver.ubuntu.com:80
keyserver hkps://keys.openpgp.org
keyserver hkp://keyserver.pgp.com
```

Running the command `gpg --verbose --recv-keys 72D7468F` gave me the following:

```
gpg: keybox '/tmp/tmp-gnupg/pubring.kbx' created
gpg: no running Dirmngr - starting '/usr/bin/dirmngr'
gpg: waiting for the dirmngr to come up ... (5s)
gpg: connection to dirmngr established
```
And it just hangs there.

`/tmp/tmp-gnupg/dirmngr.log`:

```log
2022-12-20 06:52:03 dirmngr[8928.0] permanently loaded certificates: 125
2022-12-20 06:52:03 dirmngr[8928.0]     runtime cached certificates: 0
2022-12-20 06:52:03 dirmngr[8928.0]            trusted certificates: 125 (124,0,0,1)
2022-12-20 06:52:03 dirmngr[8928.0] failed to open cache dir file '/tmp/tmp-gnupg/crls.d/DIR.txt': No such file or directory
2022-12-20 06:52:03 dirmngr[8928.0] creating directory '/tmp/tmp-gnupg/crls.d'
2022-12-20 06:52:03 dirmngr[8928.0] new cache dir file '/tmp/tmp-gnupg/crls.d/DIR.txt' created
2022-12-20 06:52:03 dirmngr[8928.6] resolve_dns_addr for 'keyserver.ubuntu.com': '162.213.33.9'
2022-12-20 06:52:03 dirmngr[8928.6] resolve_dns_addr for 'keyserver.ubuntu.com': '162.213.33.8'
2022-12-20 06:52:03 dirmngr[8928.6] number of system provided CAs: 124
```

If I change my `dirmngr.conf` to the following and run it again, it imports it successfully (i.e. remove the `:80` on the ubuntu line):

```
disable-ipv6
log-file /tmp/tmp-gnupg/dirmngr.log
keyserver hkp://keyserver.ubuntu.com
keyserver hkps://keys.openpgp.org
keyserver hkp://keyserver.pgp.com
```

```
$ gpg --verbose --recv-keys 72D7468F
gpg: keybox '/tmp/tmp-gnupg/pubring.kbx' created
gpg: no running Dirmngr - starting '/usr/bin/dirmngr'
gpg: waiting for the dirmngr to come up ... (5s)
gpg: connection to dirmngr established
gpg: data source: http://162.213.33.9:11371
gpg: armor header: Comment: Hostname:
gpg: armor header: Version: Hockeypuck 2.1.0-189-g15ebf24
gpg: pub  rsa4096/34365D9472D7468F 2021-04-19  HashiCorp Security (hashicorp.com/security) <security@hashicorp.com>
gpg: /tmp/tmp-gnupg/trustdb.gpg: trustdb created
gpg: using pgp trust model
gpg: Note: signature key B0B441097685B676 expired Wed 20 Apr 2022 12:08:32 AM UTC
gpg: key 34365D9472D7468F: public key "HashiCorp Security (hashicorp.com/security) <security@hashicorp.com>" imported
gpg: no running gpg-agent - starting '/usr/bin/gpg-agent'
gpg: waiting for the agent to come up ... (5s)
gpg: connection to agent established
gpg: Total number processed: 1
gpg:               imported: 1
```

But I've tried each of the keyservers independently and found issues with all of them.

- **hkp://keyserver.pgp.com:** seems to time out.

```
gpg: keyserver receive failed: Connection timed out

2022-12-20 07:05:09 dirmngr[12791.6] resolve_dns_addr for 'keyserver.pgp.com': '35.169.187.251'
2022-12-20 07:05:09 dirmngr[12791.6] resolve_dns_addr for 'keyserver.pgp.com': '18.206.36.226'
2022-12-20 07:05:09 dirmngr[12791.6] number of system provided CAs: 124
2022-12-20 07:05:24 dirmngr[12791.6] can't connect to '18.206.36.226': Connection timed out
2022-12-20 07:05:24 dirmngr[12791.6] error connecting to 'http://18.206.36.226:11371': Connection timed out
2022-12-20 07:05:24 dirmngr[12791.6] selecting a different host due to a timeout
2022-12-20 07:05:39 dirmngr[12791.6] can't connect to '18.206.36.226': Connection timed out
2022-12-20 07:05:39 dirmngr[12791.6] error connecting to 'http://18.206.36.226:11371': Connection timed out
2022-12-20 07:05:39 dirmngr[12791.6] selecting a different host due to a timeout
2022-12-20 07:05:54 dirmngr[12791.6] can't connect to '18.206.36.226': Connection timed out
2022-12-20 07:05:54 dirmngr[12791.6] error connecting to 'http://18.206.36.226:11371': Connection timed out
2022-12-20 07:05:54 dirmngr[12791.6] selecting a different host due to a timeout
2022-12-20 07:06:09 dirmngr[12791.6] can't connect to '35.169.187.251': Connection timed out
2022-12-20 07:06:09 dirmngr[12791.6] error connecting to 'http://35.169.187.251:11371': Connection timed out
2022-12-20 07:06:09 dirmngr[12791.6] command 'KS_GET' failed: Connection timed out
```

- **hkps://keys.openpgp.org:** fails with the short format (works just fine with `C874011F0AB405110D02105534365D9472D7468F`).

```
gpg: data source: https://keys.openpgp.org:443
gpg: keyserver receive failed: No data

2022-12-20 07:12:03 dirmngr[14298.6] error accessing 'https://keys.openpgp.org:443/pks/lookup?op=get&options=mr&search=0x72D7468F': http status 400
2022-12-20 07:12:03 dirmngr[14298.6] command 'KS_GET' failed: No data
```

I've also tried getting the key via curl from the ubuntu keyserver on port 80, and it worked just fine:
`$ curl 'http://162.213.33.9:80/pks/lookup?op=get&options=mr&search=0x72D7468F'`

I don't know what can be done to fix this, though since it seems to be behaving differently for everyone. I just wanted to share my findings.

**Comment by nabil-becker** [comment]
I am also continuing to experience the issue. Not sure what information or help I can provide to resolve the issue.

**Comment by DevSecNinja** [comment]
Reopening this issue for others, but I can't reproduce it anymore. Not sure why.

**Comment by nabil-becker** [comment]
If it helps ive tried different base images (the latest ubuntu, as well as ubuntu:jammy-20221101)

Im also experiencing this with the ruby feature, so its not just terraform.

**Comment by nabil-becker** [comment]
I was able to work around the issue. 

I created a folder in the `.devcontainers `folder, called it `0terraform`.

I then copied the devcontainer-feature.json and install.sh files for the terraform feature into that folder and updated the devcontainer.json to reference it.

![image](https://user-images.githubusercontent.com/42244256/208779307-f594dfee-9dca-45a6-a2bc-a07c77db12e7.png)

![image](https://user-images.githubusercontent.com/42244256/208779398-d902819d-8843-4d12-9b39-5598e1ce664d.png)

It looks like at line 66 of the `install.sh`, it calls the `get_common_setting `function which looks like it downloads an env file (https://aka.ms/vscode-dev-containers/script-library/settings.env) and applies it.

I added lines to override the GPG_KEY_SERVERS variable, removing the :80 for the ubuntu keyserver, which resolved the issue.

![image](https://user-images.githubusercontent.com/42244256/208779539-d4c0145c-9821-426b-9a69-17c2f28fc50c.png)

Is there anyway we can update the https://aka.ms/vscode-dev-containers/script-library/settings.env and remove the :80 reference?

**Comment by samruddhikhandale** [comment]
@nabil-becker Thanks for digging into this! 

It's very strange that GPG key gets downloaded if you don't specify a port for Ubuntu Key server. 
The Ubuntu Key server is always run on port 80 to reduce issues with firewall. I can't find an article which says otherwise.

@nabil-becker Is there a way for you to check if port 80 is blocked on your machine or if VPN is interfering with it?

**Comment by samruddhikhandale** [comment]
> Is there anyway we can update the https://aka.ms/vscode-dev-containers/script-library/settings.env and remove the :80 reference?

I could help remove the port 80 suffix from `GPG_KEY_SERVERS ` for the Features, but I'd like to confirm if it solves the issue for everyone and doesn't case another regression.

Can I get a 👍  if the following dev config works for you?

```jsonc

{
    "image": "mcr.microsoft.com/devcontainers/base:jammy",
    "features": {
        "ghcr.io/samruddhikhandale/test-terraform/terraform:1": {}
    }
}

```

I have created a temporary `terraform` Feature on my personal repo which removes `:80`

https://github.com/samruddhikhandale/test-terraform/blob/main/src/terraform/install.sh#L29

**Comment by samruddhikhandale** [comment]
Created a draft PR - https://github.com/devcontainers/features/pull/353

**Comment by nabil-becker** [comment]
Your change seems to be working, thank you

**Comment by samruddhikhandale** [comment]
> Created a PR - #353
> Once this one gets merged, I'll update other Features as well.

Merged the PR, closing as resolved. ✨

**Comment by nabil-becker** [comment]
Thank you. Im also getting the same issue with the ruby feature. Do we need a new issue opened for that?

**Comment by samruddhikhandale** [comment]
> Do we need new issues opened for that?

You may, but I have already started working on updating all other Features. I would report here once completed. 
Shouldn't take more than an hour.

**Comment by samruddhikhandale** [comment]
> > Do we need new issues opened for that?
> 
> You may, but I have already started working on updating all other Features. I would report here once completed. Shouldn't take more than an hour.

PR 👉  https://github.com/devcontainers/features/pull/354

**Comment by seBasKov** [comment]
Hi samruddhikhandale, I noticed after your latest change my terraform feature install hangs at downloading the GPG key. I was referring to major version 1 as a result it picks up your change 1.1.1.

My issue is solved using version 1.1.0 again.

**Comment by samruddhikhandale** [comment]
Looks like for some folks `keyserver hkp://keyserver.ubuntu.com` works, but for others `keyserver hkp://keyserver.ubuntu.com:80` does 😅 
The Feature probably needs to be updated to support both. 👍

**Comment by alexandreczg** [comment]
Hey, this is still happening as of today. 

devcontainer: 
```
"name": "Python 3 - Terraform",
	// Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
	"image": "mcr.microsoft.com/devcontainers/python:0-3.10",
	"features": {
		"ghcr.io/devcontainers/features/terraform:1": {},
		"ghcr.io/devcontainers-contrib/features/terraform-docs:1": {},
		"ghcr.io/devcontainers/features/git:1": {
			"version": "os-provided"
		}
	}
```
output error:
```
[2023-05-12T18:30:03.436Z] 
#0 1.711 Downloading terraform...
#0 2.608 (*) Downloading GPG key...
#0 32.67 gpg: keyserver receive failed: Server indicated a failure
#0 32.67 (*) Failed getting key, retring in 10s...
#0 42.67 (*) Downloading GPG key...
#0 72.71 gpg: keyserver receive failed: Server indicated a failure
#0 72.71 (*) Failed getting key, retring in 10s...
#0 82.71 (*) Downloading GPG key...
#0 112.7 gpg: keyserver receive failed: Server indicated a failure
#0 112.7 (*) Failed getting key, retring in 10s...
#0 122.7 (*) Downloading GPG key...
#0 152.8 gpg: keyserver receive failed: Server indicated a failure
#0 152.8 (*) Failed getting key, retring in 10s...
#0 162.8 (*) Downloading GPG key...
[2023-05-12T18:30:03.436Z] 
[2023-05-12T18:30:03.436Z] #0 192.8 gpg: keyserver receive failed: Server indicated a failure
#0 192.8 (*) Failed getting key, retring in 10s...
[2023-05-12T18:30:03.436Z] #0 202.8 (!) Failed to get gpg key.
#0 202.8 ERROR: Feature "Terraform, tflint, and TFGrunt" (ghcr.io/devcontainers/features/terraform) failed to install! Look at the documentation at https://github.com/devcontainers/features/tree/main/src/terraform for help troubleshooting this error.
[2023-05-12T18:30:03.436Z] ------
[2023-05-12T18:30:03.436Z] Dockerfile.extended:42
--------------------
  41 |     
  42 | >>> RUN --mount=type=bind,from=dev_containers_feature_content_source,source=terraform_1,target=/tmp/build-features-src/terraform_1 \
[2023-05-12T18:30:03.437Z]   43 | >>>     cp -ar /tmp/build-features-src/terraform_1 /tmp/dev-container-features \
  44 | >>>  && chmod -R 0755 /tmp/dev-container-features/terraform_1 \
  45 | >>>  && cd /tmp/dev-container-features/terraform_1 \
  46 | >>>  && chmod +x ./devcontainer-features-install.sh \
  47 | >>>  && ./devcontainer-features-install.sh \
  48 | >>>  && rm -rf /tmp/dev-container-features/terraform_1
  49 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c cp -ar /tmp/build-features-src/terraform_1 /tmp/dev-container-features  && chmod -R 0755 /tmp/dev-container-features/terraform_1  && cd /tmp/dev-container-features/terraform_1  && chmod +x ./devcontainer-features-install.sh  && ./devcontainer-features-install.sh  && rm -rf /tmp/dev-container-features/terraform_1" did not complete successfully: exit code: 1

```

**Comment by samruddhikhandale** [comment]
Hi @alexandreczg  👋 

This issue looks similar to https://github.com/devcontainers/features/issues/435. @alexandreczg are you trying to run the dev container in a proxy environment? If so, can you add the `httpProxy` to the [terraform's Feature option](https://github.com/devcontainers/features/blob/main/src/terraform/README.md#options)

Also, the [smoke test](https://github.com/devcontainers/features/actions/runs/4961980894) passes for the terraform Feature. Hence, I suspect an issue with the proxy.

**Comment by alexandreczg** [comment]
Hi @samruddhikhandale , thank you for the quick reply. 

I'm not sure what a "proxy environment" is in this context. I tried running the same container in windows and WSL2. Same outcome. The only thing I can think of is my company's VPN. What should I put in the httpProxy option? It requires a string.

**Comment by alexandreczg** [comment]
Hi again @samruddhikhandale ,

Can confirm, the issue is the VPN. Tried without and built it without any issues.

**Comment by samruddhikhandale** [comment]
@alexandreczg setting `httpProxy` will allow you to start this container with VPN enabled. It is of this format `=""http://<username>:<password>@<proxy_server_addr>:<proxy_port>""` (or eg. `http://corporate.proxy.test:8765`)

**Comment by alexandreczg** [comment]
Thank you, 

Dumb question, but how do I find my proxy settings? I am very unfamiliar with this. Any help would be appreciated!
